src/core/Form/Select/MultiSelect/MultiSelect/MultiSelect.tsx

```diff
@@ -61,7 +62,7 @@ export interface MultiSelectProps<T extends MultiSelectData> {
   optionalText?: string;
   /** Hint text to be shown below the label */
   hintText?: string;
-  /** Event that is fired when item selections change. Not fired when in controlled state. */
+  /** Event that is fired when item selections change */
   onItemSelectionsChange?: (selectedItems: Array<T>) => void;
```

Comment:
Is there a reason this array is only of type T and not T & MultiSelectData like most other selected items related arrays? Not changed in this PR but just hit my eye.

Reply:
Yes, I think it should be the same type as used everywhere else with items. 

Changed it for both SingleSelect & MultiSelect